ipfsconn/ipfshttp/ipfshttp.go

```diff
@@ -612,6 +617,48 @@ func (ipfs *Connector) RepoStat(ctx context.Context) (*api.IPFSRepoStat, error)
 	return &stats, nil
 }
 
+// RepoGC performs a garbage collection sweep on the repo.
+func (ipfs *Connector) RepoGC(ctx context.Context) (*api.RepoGC, error) {
+	ctx, span := trace.StartSpan(ctx, "ipfsconn/ipfshttp/RepoGC")
+	defer span.End()
+
+	id, err := ipfs.ID(ctx)
+	if err != nil {
+		logger.Error(err)
+		return nil, err
+	}
+	repoGC := api.RepoGC{
+		Peer: id.ID,
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, ipfs.config.IPFSRequestTimeout)
+	defer cancel()
+
+	res, err := ipfs.doPostCtx(ctx, ipfs.client, ipfs.apiURL(), "repo/gc", "", nil)
+	if err != nil {
+		logger.Error(err)
+		return &repoGC, err
+	}
+	defer res.Body.Close()
+
+	dec := json.NewDecoder(res.Body)
+	for {
+		resp := ipfsRepoGCResp{}
+		if err := dec.Decode(&resp); err == io.EOF {
```

Comment:
see https://github.com/ipfs/ipfs-cluster/blob/master/ipfsconn/ipfshttp/ipfshttp.go#L413 for a slightly improved error handling.

Reply:
It now uses the suggested error handling